thorlcr/thorutil/thbuf.cpp

```diff
@@ -879,8 +879,8 @@ class CCompressedSpillingRowStream: public CSimpleInterfaceOf<ISmartRowBuffer>,
         rowcount_t currentNextOutputRow = nextOutputRow.load();
         trace("WRITE: Flushed to disk. nextOutputRow = %" RCPF "u", currentNextOutputRow);
         outputStream->flush();
+        currentOwnedOutputFile->noteSize(currentOutputIFileIO->getStatistic(StSizeDiskWrite));
         currentTempFileSize += pendingFlushToDiskSz;
-        currentOwnedOutputFile->noteSize(currentTempFileSize);
         pendingFlushToDiskSz = 0;
         if (currentTempFileSize > options.tempFileGranularity)
```

Comment:
@jakesmith should this threshold be on compressed or uncompressed size?  If compressed you could change to say 
currentTempFileSize = currentOutputIFileIO->getStatistic(StSizeDiskWrite)

Reply:
@ghalliday - please see change (new commit)